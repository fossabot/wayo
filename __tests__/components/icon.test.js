import 'babel-polyfill';
import { shallowMount,createLocalVue, mount } from '@vue/test-utils';
import Icon from '../../src/components/icon';

describe('SnapShot Cases',()=>{
  test('icon',()=>{
    const Wrapper = mount(Icon);
    expect(Wrapper).toMatchSnapshot();
  });  
});

describe('Logic Cases',()=>{
  const LocalVue = createLocalVue();

  const Wrapper = shallowMount(Icon, {
    localVue: LocalVue,
    propsData: {
      name: 'bus',
      color: '#ff0000',
      size: 12
    }
  });

  test('class',()=>{
    expect(Wrapper.classes()).toContain('wayo-icon-bus');
  }); 

  test('styles',()=>{
    expect(Wrapper.vm.styles).toContain('color: #ff0000;');
    expect(Wrapper.vm.styles).toContain('font-size: 12px;');
  });  
});