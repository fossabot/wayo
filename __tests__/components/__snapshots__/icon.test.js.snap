// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnapShot Cases icon 1`] = `<i class="wayo-icon wayo-icon-"></i>`;
